src/test/java/decodex/data/DataManagerTest.java

```diff
@@ -0,0 +1,30 @@
+package decodex.data;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+class DataManagerTest {
+
+    @Test
+    public void resetCurrentData_noUpdates_originalData() {
+        Data originalData = new Data("hi");
+        DataManager testManager = new DataManager(originalData);
+
+        testManager.resetToOriginalData();
+
+        assertTrue(testManager.getCurrentData() == originalData);

```

Comment:
Using == is ambiguous as it may be checking object reference instead. Would suggest checking equality by the raw bytes, or overriding the equals() method.